### PR TITLE
Aave V3 retro funding - Review

### PR DESCRIPTION
**Pull request by marczeller**
This AIP is meant for a guardian proposal creation on AVAX.

Please consider reviewing the Payload.

### CODE REVIEW

File: src/AaveV3AvaxretrofundingPayload_20230626/AaveV3AvaxretrofundingPayload_20230626_Test.t.sol [hunk 1]
```diff
@@ -0,0 +1,83 @@
...
+    assertApproxEqAbs(aaveCoAfter.aAvaWAVAX - aaveCoBefore.aAvaWAVAX, AMOUNT_aAvaWAVAX, 1);
+
+    assertApproxEqAbs(collectorBefore.avWAVAX - collectorAfter.avWAVAX, AMOUNT_avWAVAX, 1);
+    assertApproxEqAbs(aaveCoAfter.avWAVAX - aaveCoBefore.avWAVAX, AMOUNT_avWAVAX, 1);
+
+    assertApproxEqAbs(collectorBefore.aAvaWBTC - collectorAfter.aAvaWBTC, AMOUNT_aAvaWBTC, 1);
+    assertApproxEqAbs(aaveCoAfter.aAvaWBTC - aaveCoBefore.aAvaWBTC, AMOUNT_aAvaWBTC, 1);
+
+    assertApproxEqAbs(collectorBefore.avWBTC - collectorAfter.avWBTC, AMOUNT_avWBTC, 1);
+    assertApproxEqAbs(aaveCoAfter.avWBTC - aaveCoBefore.avWBTC, AMOUNT_avWBTC, 1);
```
Comment (rustboyar) [inline src/AaveV3AvaxretrofundingPayload_20230626/AaveV3AvaxretrofundingPayload_20230626_Test.t.sol:81]: you can also check that AAVECO_ADDRESS received the funds

File: src/AaveV3AvaxretrofundingPayload_20230626/AaveV3AvaxretrofundingPayload_20230626_Test.t.sol [hunk 2]
```diff
@@ -0,0 +1,83 @@
...
+contract AaveV3AvaxretrofundingPayloadTest is Test {
+  struct TokenBalances {
+    uint256 aAvaWAVAX;
+    uint256 avWAVAX;
+    uint256 aAvaWBTC;
+    uint256 avWBTC;
+  }
+
+  address public constant COLLECTOR_ADDRESS = address(AaveV3Avalanche.COLLECTOR);
+  address public constant AAVECO_ADDRESS = 0xa8F8E9c54e099c4fADB797f5196E07ce484824aF;
```
[the remark this answers is not in the document]
Comment (Zer0dot) [reply src/AaveV3AvaxretrofundingPayload_20230626/AaveV3AvaxretrofundingPayload_20230626_Test.t.sol:20]: I think it's good not to, ensures that if the test passes, the addresses are consistent, no?

[the remark this answers is not in the document]
Comment (rustboyar) [reply src/AaveV3AvaxretrofundingPayload_20230626/AaveV3AvaxretrofundingPayload_20230626_Test.t.sol:20]: makes sense, agree

**Review by rustboyar — APPROVED** [review]

File: src/AaveV3AvaxretrofundingPayload_20230626/AaveV3AvaxretrofundingPayload_20230626.sol [hunk 3]
```diff
@@ -0,0 +1,35 @@
...
+/**
+ * @title AAVE V3 AVAX Retro funding proposal
+ * @author @marczeller - Aave-Chan Initiative
+ * Governance Forum Post: https://governance.aave.com/t/arc-aave-v3-retroactive-funding/9250
+ * Snapshot: https://snapshot.org/#/aave.eth/proposal/bafkreia45d5m6y3o4b476ykp6q65v3zlecxvz6h57tdfhowqm5wb5nzdvq
+ */
+contract AaveV3AvaxretrofundingPayload is IProposalGenericExecutor {
+    address public constant AAVECO_ADDRESS = 0xa8F8E9c54e099c4fADB797f5196E07ce484824aF;
+    address public constant aAvaWAVAX = AaveV3AvalancheAssets.WAVAX_A_TOKEN;
+    address public constant avWAVAX = 0xDFE521292EcE2A4f44242efBcD66Bc594CA9714B;
```
[the remark this answers is not in the document]
Comment (marczeller) [reply src/AaveV3AvaxretrofundingPayload_20230626/AaveV3AvaxretrofundingPayload_20230626.sol]: done

File: src/AaveV3AvaxretrofundingPayload_20230626/AaveV3AvaxretrofundingPayload_20230626.sol [hunk 4]
```diff
@@ -0,0 +1,35 @@
...
+    address public constant aAvaWBTC = AaveV3AvalancheAssets.WBTCe_A_TOKEN;
+    address public constant avWBTC = 0x686bEF2417b6Dc32C50a3cBfbCC3bb60E1e9a15D;
```
[the remark this answers is not in the document]
Comment (marczeller) [reply src/AaveV3AvaxretrofundingPayload_20230626/AaveV3AvaxretrofundingPayload_20230626.sol]: done

**Pull request merged**